### PR TITLE
add vm discovery for cloudagent

### PR DESCRIPTION
**Pull request by katl-msft**

### CODE REVIEW

File: services/compute/compute.go [hunk 1]
```diff
@@ -1160,3 +1160,12 @@ type ProxyConfiguration struct {
 	// Alternative CA cert to use for connecting to proxy server
 	TrustedCa *string `json:"trustedca,omitempty"`
 }
+
+type VirtualMachineDiscovery struct {
```
Comment (mduppre) [inline services/compute/compute.go:1164]: As we discussed yesterday, I think we don't need this struct after all as we're just using the `VirtualMachine` struct

File: services/compute/virtualmachine/virtualmachine.go [hunk 2]
```diff
@@ -457,12 +457,12 @@ func (c *client) getVirtualMachine(vm *wssdcloudcompute.VirtualMachine, group st
 		vmtype = compute.StackedControlPlane
 	}
 	return &compute.VirtualMachine{
-		Name:                     &vm.Name,
-		ID:                       &vm.Id,
-		Tags:                     getComputeTags(vm.GetTags()),
+		Name: &vm.Name,
```
Comment (mduppre) [inline services/compute/virtualmachine/virtualmachine.go:460]: Would you mind reverting the whitespace changes? Obviously very low priority, just will make our overall PR easier to read

**Review by mduppre — APPROVED** [review]
Looks good :-)

File: services/compute/virtualmachine/virtualmachine.go (hunk 2 above)
Comment (neerajdixit-msft2) [reply services/compute/virtualmachine/virtualmachine.go:460]: I think these are done automatically by `make fmt` according to go rules so better keep it as is.

Comment (mduppre) [reply services/compute/virtualmachine/virtualmachine.go:460]: Makes sense!

File: services/compute/compute.go (hunk 1 above)
Comment (neerajdixit-msft2) [reply services/compute/compute.go:1164]: @mduppre I think we will need this structure if we want to change the API function signature.

Comment (mduppre) [reply services/compute/compute.go:1164]: yeah, you're right, I wrote this comment before we chatted about the API function signature. So let's leave it and I'll look into converting the objects

**Pull request merged**